src/app/loadout-drawer/loadout-drawer-reducer.test.ts

```diff
@@ -0,0 +1,574 @@
+    loadout = addItem(defs, item2, true)(loadout);
+
+    expect(loadout.items).toMatchObject([
+      {
+        equip: false,
+        id: item.id,
+      },
+      {
+        equip: true,
+        id: item2.id,
+      },
+    ]);
+  });
+
+  it('can switch an equipped item to unequipped if added again with the equip flag set', () => {
+    const item = items[0];
+
+    let loadout = addItem(defs, item, true)(emptyLoadout);
+    loadout = addItem(defs, item, false)(loadout);
+
+    expect(loadout.items).toMatchObject([
+      {
+        equip: false,
+        id: item.id,
+      },
+    ]);
+  });
+
+  it('can switch an unequipped item to equipped if added again with the equip flag set', () => {
+    const item = items[0];
+
+    let loadout = addItem(defs, item, false)(emptyLoadout);
+    loadout = addItem(defs, item, true)(loadout);
+
+    expect(loadout.items).toMatchObject([
+      {
+        equip: true,
+        id: item.id,
+      },
+    ]);
+  });
+
+  it('handles duplicates even if the new version is a reshaped version of the one saved in the loadout', () => {
+    const item = items.find((i) => i.crafted)!;
+
+    let loadout = addItem(defs, item, true)(emptyLoadout);
+    loadout = addItem(defs, { ...item, id: '1234' }, false)(loadout);
+
+    expect(loadout.items).toMatchObject([
+      {
+        equip: false,
+        id: '1234',
+      },
+    ]);
+  });
+
+  it('fills in socket overrides when adding a subclass', () => {
+    const subclass = items.find((i) => i.bucket.hash === BucketHashes.Subclass)!;
+
+    let loadout = addItem(defs, subclass)(emptyLoadout);
+
+    expect(loadout.items[0].socketOverrides).toBeDefined();
+  });
+
+  it('replaces the existing subclass when adding a new one', () => {
+    const [subclass, subclass2] = items.filter((i) => i.bucket.hash === BucketHashes.Subclass);
+
+    let loadout = addItem(defs, subclass)(emptyLoadout);
+    loadout = addItem(defs, subclass2)(loadout);
+
+    expect(loadout.items).toMatchObject([
+      {
+        id: subclass2.id,
+      },
+    ]);
+  });
+
+  it('does nothing if the item cannot be in a loadout', () => {
+    const invalidItem = store.items.find((i) => !itemCanBeInLoadout(i))!;
+    expect(invalidItem).toBeDefined();
+
+    let loadout = addItem(defs, invalidItem)(emptyLoadout);
+
+    expect(loadout.items).toEqual([]);
+  });
+
+  it('does nothing if the item is for the wrong class', () => {
+    const invalidItem = allItems.find((i) => !isClassCompatible(i.classType, DestinyClass.Hunter))!;
+    expect(invalidItem).toBeDefined();
+
+    let loadout = addItem(defs, invalidItem)(emptyLoadout);
+
+    expect(loadout.items).toEqual([]);
+  });
+
+  it('does nothing if the item cannot be in a loadout', () => {
+    const invalidItem = allItems.find((i) => !itemCanBeInLoadout(i))!;
+    expect(invalidItem).toBeDefined();
+
+    let loadout = addItem(defs, invalidItem)(emptyLoadout);
+
+    expect(loadout.items).toEqual([]);
+  });
```

Comment:
This test seems to be duplicated (only difference is grabbing the item from `allItems` instead of `store`).